armi/reactor/blueprints/reactorBlueprint.py

```diff
@@ -74,25 +74,29 @@ def __init__(self, name=None, gridName=None, origin=None):
 
     @staticmethod
     def _resolveSystemType(typ: str):
-        # avoid circular import, though ideally reactors shouldnt depend on blueprints!
-        from armi.reactor import reactors
-        from armi.reactor import assemblyLists
-
-        # TODO: This is far from the the perfect place for this; most likely plugins
-        # will need to be able to extend it. Also, the concept of "system" will need to
-        # be well-defined, and most systems will probably need their own, rather
-        # dissimilar Blueprints. We may need to break with yamlize unfortunately.
-        _map = {"core": reactors.Core, "sfp": assemblyLists.SpentFuelPool}
-
-        try:
-            cls = _map[typ]
-        except KeyError:
-            raise ValueError(
-                "Could not determine an appropriate class for handling a "
-                "system of type `{}`. Supported types are {}.".format(typ, _map.keys())
-            )
-
-        return cls
+        # Loop over all plugins that could be attached and determine if any
+        # tell us how to build a specific systems attribute. Sub-optimial
+        # as this check is called for each system (e.g., core, spent fuel pool).
+        # It is assumed that the number of systems is currently low enough to justify
+        # this structure.
+
+        manager = armi.getPluginManagerOrFail()
+
+        # Only need this to handle the case we don't find the grid we expect
+        seen = set()
+        for options in manager.hook.defineSystemGridBuilders():
+            for key, builder in options.items():
+                # Take the first match we find. This would allow other plugins to
+                # define a new core builder before finding those defined by the
+                # ReactorPlugin
+                if key == typ:
+                    return builder
+                seen.add(key)
+
+        raise ValueError(
+            "Could not determine an appropriate class for handling a "
+            "system of type `{}`. Supported types are {}.".format(typ, sorted(seen))
+        )
```

Comment:
@drewj-usnctech This is a non-trivial change.  And while it _looks_ fine to me, it is a bit hard for me to tell if there are edge cases where this would fail. What would help me understand/grok the changes if if we had a unit test to cover this new/re-written method. Does that make sense?

The obvious place would seem to be `armi/reactor/blueprints/tests/test_reactorBlueprints.py`.

Side Note: Again, I'm sorry it took so long for me to get to this PR.  I wasn't ignoring you, I've just been very busy and when I saw the little red "X" next to your PR I thought it wasn't ready yet.  So my triage procedure was faulty. My  bad.

Reply:
@john-science you're totally fine on the delay. and yes, this is a non-trivial change where a unit test would help. I'm not sure how to handle a unit test where we would add something to the plugin manager returned by `armi.getPluginMnagerOrFail`. The testable way might be to make this rely on another method that, given any plugin manager, performs this lookup. Then we don't have to touch `armi.getPluginManagerOrFail`

Possible path: refactor the call signature to accept a string `typ` and the plugin manager, then test that "private" method? How do you feel about that?